src/modules/page.js

```diff
@@ -176,13 +176,29 @@ function getSubmitButton(form) {
     return null;
   }
 
-  return Array.prototype.slice.call(buttons, buttons.length - 1, buttons.length)[0];
+  let submitButtonPredicates = [
+    // explicit submit type
+    (button) => button.getAttribute("type") === "submit",
+    // the browser interprets an unset or invalid type as submit
+    (button) => !Array.prototype.includes
+                                .call(["submit", "button"], button.getAttribute("type")),
+    // assume that last button in form performs submission via javascript
+    (button, index, arr) => index + 1 === arr.length
+  ];
+
+  for (let predicate of submitButtonPredicates) {
+    button = Array.prototype.find.call(buttons, predicate);
```

Comment:
If I'm not mistaken, a variable declaration (`let button = ...`) is missing here.

Reply:
Oh, good catch, fixed.